overlord/state/task.go

```diff
@@ -522,6 +522,9 @@ func (ts *TaskSet) AddTask(task *Task) {
 // MarkEdge marks the given task as a specific edge. Any pre-existing
 // edge mark will be overridden.
 func (ts *TaskSet) MarkEdge(task *Task, edge TaskSetEdge) {
+	if task == nil {
```

Comment:
A unit test would be nice.

Reply:
Added. It actually uncovered a mistake in panic string 😯

Reply:
Hah, right `%q` missing format.